builtins: add json_valid

eaarranz opened the pull request:
Created the json_valid builtin function to validate whether a String is a valid JSON or not.

Resolves: #71453

rafiss reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/71725)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @eaarranz and @smcvey)

___
*[pkg/sql/logictest/testdata/logic_test/json, line 867 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/71725#-MmP94Yg33JDhBnOAlh9:-MmP94Yg33JDhBnOAlhA:bs58p0a) ([raw file](https://github.com/cockroachdb/cockroach/blob/8537b4bc9478ff106cfd546e2067097ad86f912c/pkg/sql/logictest/testdata/logic_test/json#L867)):*
> ```
> 
> query B
> SELECT json_valid('[{"bar": 1}, {"bar": 2}]')
> ```

nit: could you add one more test for `json_valid(NULL)`

hm, i'm actually not sure what the desired behavior should be for that. i'm open to either returning NULL or true for it. but whatever it is, it should be well-defined.


<!-- Sent from Reviewable.io -->

otan commented:
can you fold your commit into one, as per the blathers comment?
```
git rebase -i HEAD~2 # and then use "f" on the second commit
git log # check it's just one commit
git push --force
```

eaarranz committed:
builtins: add json_valid

Created the json_valid builtin function to validate whether a String is a valid JSON or not.

- Added a test to ensure that json_valid(NULL) returns NULL
- Removed the unused error returned from jsonValidate

Resolves: #71453
Release note: None

otan reviewed (approved):
cheers

bors r+
